dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java

```diff
@@ -333,8 +337,18 @@ public <T> void reExport(Exporter<T> exporter, URL newInvokerUrl) {
      */
     @SuppressWarnings("unchecked")
     public <T> void reExport(final Invoker<T> originInvoker, URL newInvokerUrl) {
-        String key = getCacheKey(originInvoker);
-        ExporterChangeableWrapper<T> exporter = (ExporterChangeableWrapper<T>) bounds.get(key);
+        String providerUrlKey = getProviderUrlKey(originInvoker);
+        String registryUrlKey = getRegistryUrlKey(originInvoker);
+        Map<String, ExporterChangeableWrapper<?>> registryMap = bounds.get(providerUrlKey);
```

Comment:
What's the purpose of adding an extra `registryUrl` layer for exporters?

Is it because we need to do `reExport` for a single registry at one time?

Reply:
In the scenario of multiple registries(e.g. zk:2181, zk:2182), the providerUrlKeys are the same, but invokers are different.